encoding/ssz/htrutils.go

```diff
@@ -141,19 +142,21 @@ func WithdrawalSliceRoot(withdrawals []*enginev1.Withdrawal, limit uint64) ([32]
 	return MixInLength(bytesRoot, bytesRootBufRoot), nil
 }
 
-func transactionRoot(tx []byte) ([32]byte, error) {
-	chunkedRoots, err := PackByChunk([][]byte{tx})
+// ByteSliceRoot is a helper func to merkleize an arbitrary List[Byte, N]
+// this func runs Chunkify + MerkleizeVector
+// max length is dividable by 32 ( root length )
+func ByteSliceRoot(slice []byte, maxLength uint64) ([32]byte, error) {
```

Comment:
This function is a little odd, we should be able to hash arbitrary byte slices with the exact same algorithm, why would we restrict to chunked inputs? (multiple of 32 that is)

Reply:
so we should add a new function instead of BitwiseMerkleize for this perhaps? that is chunked.